### DISCUSSION TITLE
CustomTarget: Is there a feature to limit target size when the image is large?

### DISCUSSION THREAD

**Discussion by annotium** [coil-kt/coil#825] [General]
I use Coil to load the image from device to custom target. But sometimes I got the large image and have problem with memory.

```kt
val request = ImageRequest.Builder(context)
                .data(uri)
                .build()
val result = context.imageLoader.execute(request)
```

From Glide it has a very good feature is `centerInside()`. That way loaded image won’t expand beyond the given size.
Is there a similar feature in Coil?

**Comment by colinrtwhite** [comment]
You can set the size you want to load the drawable at using `ImageRequest.Builder.size`.

**Reply by annotium** [reply]
Thanks @colinrtwhite , maybe you misunderstand me.

I load an image without knowing its size to the custom target.

e.g I have original image with size (4000x2000). But I only want to load it with a maximum size (2000x2000). So in Glide I prepare a size (2000) and used `centerInside()`, so the result bitmap has size (2000x1000).

```kt

// code
Glide.with(context)
     .load(IMAGE_URL)
     .size(2000)
     .centerInside()
     .into(customTarget)

// use the bitmap
```

If I load smaller images, e.g 1000x500, the the output bitmap is still 1000x500, it only effects with images have larger size than the given size. 

Is there a same effect if I only set the size in Coil?

**Reply by annotium** [reply]
Finally found, for people have same problem, this is a optimization

```kt
ImageRequest.Builder(context)
                .allowHardware(false)
                .size(1000)
                .scale(Scale.FIT)
                .precision(Precision.INEXACT)
                .data(uri)
                .build()
```
